samples/go/photo-index/main.go

```diff
@@ -27,21 +29,13 @@ func index() (win bool) {
 	var dbStr = flag.String("db", "", "input database spec")
 	var outDSStr = flag.String("out-ds", "", "output dataset to write to - if empty, defaults to input dataset")
 	var parallelism = flag.Int("parallelism", 16, "number of parallel goroutines to search")
+	verbose.RegisterVerboseFlags(flag.CommandLine)
 
 	flag.Usage = usage
 	flag.Parse(false)
 
-	if flag.NArg() == 0 {
-		flag.Usage()
-		return
-	}
-
-	if flag.NArg() == 0 {
```

Comment:
It's legal to have no db argument since a default might be defined in .nomsconfig. If not cfg.GetDatabase will return an error. Error reporting could be better, but my first cut here tried to allow integration with existing code with minimal changes.